src/analysis/seminorm.lean

```diff
@@ -609,6 +609,22 @@ begin
   exact (map_add_le_add p _ _).trans (add_le_add hy₁ hy₂)
 end
 
+/- Can this be done more neatly with `seminorm.ball_comp`?
+Alternatively, can this be done using/imitating the ball-addition-lemmas
+in `normed_space/pointwise`? -/
+/-- The image of a ball under addition with a singleton is another ball. -/
+lemma singleton_add_ball (p : seminorm 𝕜 E) :
+  (λ (z : E), x + z) '' p.ball y r = p.ball (x + y) r :=
+begin
+  apply le_antisymm,
+  { rintros _ ⟨_, _, hz₁z₀⟩,
+    rwa [mem_ball, ←hz₁z₀, add_sub_add_left_eq_sub] },
+  { intros z hz,
+    refine set.mem_image_iff_bex.mpr ⟨z - x, _⟩,
+    rw [mem_ball, sub_sub, add_sub, add_sub_cancel'],
+    exact ⟨hz, rfl⟩ }
```

Comment:
```suggestion
  { exact λ z hz, ⟨z - x, ⟨(p.sub_mem_ball _ _ _ _).mpr hz, add_eq_of_eq_sub' rfl⟩⟩ }
```
and add the lemma
```

lemma sub_mem_ball (p : seminorm 𝕜 E) (x₁ x₂ y : E) (r : ℝ) :
  x₁ - x₂ ∈ p.ball y r ↔ x₁ ∈ p.ball (x₂ + y) r :=
by simp_rw [mem_ball, sub_sub]
```

Reply:
Nice one! I'll use the weekend to see if I can make the other inclusion a short one-liner, as well.